VAMS 20162 news release defects

brianseek opened the pull request:
# Description

This PR address several defects with the <PressRelease> component. See more [detail in the ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20162).

## Generated description

This pull request introduces updates to the `PhoneNumber` component, refactors the `PressRelease` layout, and enhances test coverage. Key changes include adding optional props, improving testability with `data-testid` attributes, and streamlining the handling of phone numbers and media assets.

### Updates to the `PhoneNumber` Component:
* Added a new optional `testId` prop to the `PhoneNumber` component for improved testability. (`src/templates/common/phoneNumber/index.tsx`, [[1]](diffhunk://#diff-da9888787c5c2162a69df279c2086e7d30cf993198f78b47e29d212cf48548f3R47-R50) [[2]](diffhunk://#diff-da9888787c5c2162a69df279c2086e7d30cf993198f78b47e29d212cf48548f3L76-R85)

### Refactoring of the `PressRelease` Layout:
* Replaced custom button and link elements with `va-button` and `va-link` components for consistency, and added `data-testid` attributes for testing. (`src/templates/layouts/pressRelease/index.tsx`, [[1]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bL39-R101) [[2]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bR114-R122) [[3]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bL135-R132) [[4]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bL146-R146) [[5]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bL157-R160)
* Updated the handling of media contacts to skip rendering when all contact data is null and added a test for this scenario. (`src/templates/layouts/pressRelease/index.tsx`, [[1]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bL39-R101) [[2]](diffhunk://#diff-ae1dc2a62c2fc47ad6a9ce7356419154ac1bd2dd645093e51edb54769af7778bR114-R122) [[3]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL251-R263)

### Improvements to Tests:
* Replaced text-based queries with `data-testid` queries in `PressRelease` tests for more robust test cases. (`src/templates/layouts/pressRelease/index.test.tsx`, [[1]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL108-R109) [[2]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL134-R135) [[3]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL251-R263)
* Adjusted expectations in tests to account for updated phone number formatting and attributes. (`src/templates/layouts/pressRelease/index.test.tsx`, [[1]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL159-R179) [[2]](diffhunk://#diff-966b2ff1bc9fc89a75a2ff1009d389bd0f076ce6c06fda5552f12a1c2617380dL225-R212)

### Type Adjustments:
* Made the `label` property in the `PhoneNumber` type optional to align with the updated component logic. (`src/types/formatted/phoneNumber.ts`, [src/types/formatted/phoneNumber.tsL5-R5](diffhunk://#diff-bf166dc2bee1f4887a057fd15b4431264a91b6e50e1aa713253c4fba920d53dcL5-R5))

## Ticket

<!--
https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue
-->

Closes 20162

## Developer Task

- [x] PR submitted against the `main` branch of `next-build`.
- [x] Link to the issue that this PR addresses (if applicable).
- [x] Define all changes in your PR and note any changes that could potentially be breaking changes.
- [x] PR includes steps to test your changes and links to these changes in the Tugboat preview (if applicable).
- [x] Provided before and after screenshots of your changes (if applicable).
- [x] Alerted the #next-build Slack channel to request a PR review.
- [x] You understand that once approved, you are responsible for merging your changes into `main`. (Note that changes to `main` will move automatically into production.)

## Testing Steps

 - [ ] Visit https://pr1003-nn2vchgzduqq3fgsxyjfcjtdhvrklzfc.tugboat.vfs.va.gov/boston-health-care/news-releases/clinical-study-on-use-of-raman-spectroscopy-device-to-detect-skin-cancer/
 - [ ] verify that the print button and download link match prod and use the DS components, va-button for print and va-link for download
 - [ ] verify that the headings "Media contacts" and "Download media assets" are H2s
 - [ ] Visit https://pr1003-nn2vchgzduqq3fgsxyjfcjtdhvrklzfc.tugboat.vfs.va.gov/providence-health-care/news-releases/providence-va-invests-in-mental-health-services-with-new-facility/
 - [ ] verify that the entire Media contacts section is not rendered

## Screenshots
![localhost_3999_boston-health-care_news-releases_clinical-study-on-use-of-raman-spectroscopy-device-to-detect-skin-cancer_ (1)](https://github.com/user-attachments/assets/97d521fd-724e-4438-bddb-411b1ef22b51)
![localhost_3999_providence-health-care_news-releases_providence-va-invests-in-mental-health-services-with-new-facility_ (1)](https://github.com/user-attachments/assets/efa17acc-1224-4309-9564-44dec422953f)

---

## Reviewer

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository.

## Standard Checks


- [ ] Code Quality: Readabilty, Naming Conventions, Consistency, Reusability
- [ ] Test Coverage: 80% coverage
- [ ] Functionality: Change functions as expected with no additional bugs
- [ ] Performance: Code does not introduce performance issues
- [ ] Documentation: Changes are documented in their respective README.md files
- [ ] Security: Packages have been approved in the TRM


## Merging a Layout

When merging a layout, you must ensure that the content type has been turned on for `next-build` in the [.tugboat.env](../envs/.tugboat.env). This method mocks the CMS flag that must be turned on for a layout to be included in the build.

The layout component and matching resource type should be included in the [slug.tsx](../src/pages/[[...slug]].tsx), so that it can reviewed. Including a component in the slug.tsx does not mean a page will be viewable in production only on the tugboat for the branch.

When a layout is merged to main and approved for deployment, the prod CMS will turn the toggle on for the resource type. 

The status of layouts should be kept up to date inside [templates.md](../READMEs/templates.md). This includes QA progress, development progress, etc. A link should be provided for where testing can occur.

brianseek commented on src/templates/layouts/pressRelease/index.test.tsx:
```diff
@@ -156,40 +156,27 @@ describe('<pressRelease> with valid data', () => {
     const ext = screen.getByTestId('phone-3')
     const fax = screen.getByTestId('phone-4')
     expect(voice).toBeInTheDocument()
-    expect(voice).toHaveAttribute('contact', data.contacts[0].numbers[0].number)
```
Since this is switching to the shared component, we no longer need to test all of this logic here.

brianseek commented on src/templates/layouts/pressRelease/index.tsx:
```diff
@@ -2,8 +2,8 @@ import { PressRelease as FormattedPressRelease } from '@/types/formatted/pressRe
 import { LovellStaticPropsResource } from '@/lib/drupal/lovell/types'
 import { LovellSwitcher } from '@/templates/components/lovellSwitcher'
 import { formatDate } from '@/lib/utils/helpers'
-import { recordEvent } from '@/lib/analytics/recordEvent'
```
was not being used

brianseek commented on src/types/formatted/phoneNumber.ts:
```diff
@@ -2,7 +2,7 @@ import { PublishedParagraph } from '@/types/formatted/publishedEntity'
 export type PhoneNumber = PublishedParagraph & {
   extension: string
-  label: string
+  label?: string
```
label is optional since the component can also derive the label from phone type

cvalarida replied:
One of these days, I'd love to switch on strict mode so these optional properties are actually meaningful. 😮‍💨 

Thanks for being deliberate about this, though!

cvalarida commented on src/templates/common/phoneNumber/index.tsx:
```diff
@@ -73,14 +74,15 @@ export const PhoneNumber = (
   return (
     <p className={className || undefined} data-testid="phone">
-      <strong>{labelToDisplay}: </strong>
+      <strong>{`${labelToDisplay}: `}</strong>
```
Was this giving you some grief before? 🤔

brianseek replied:
It was rendering with a line break in html but not on screen which is bad for screen readers and test assertions.
![Screenshot 2025-05-01 at 9 13 54 AM](https://github.com/user-attachments/assets/9f2e2e9b-0ef7-4f7c-a13d-44ff53530f28)

cvalarida commented on src/templates/layouts/pressRelease/index.test.tsx:
```diff
@@ -131,8 +131,8 @@ describe('<pressRelease> with valid data', () => {
-    const { container } = render(<PressRelease {...data} />)
-    const linkElement = screen.queryByText(/Download press release \(PDF\)/i)
+    render(<PressRelease {...data} />)
+    const linkElement = screen.queryByTestId('pdf-version')
```
Solid improvement. 👍

cvalarida commented on src/templates/layouts/pressRelease/index.test.tsx:
```diff
@@ -248,8 +235,30 @@ describe('<pressRelease> with valid data', () => {
     expect(window.print).toHaveBeenCalled()
   })
+  test('does not render Media contacts section when the contact is archived', () => {
+    // This is the state of data at the component when the author is archived
+    const archivedAuthorContacts = [
+      {
+        id: '96668498-d442-4b55-8b14-7ec90410f418',
+        name: null,
+        description: null,
+        numbers: [
+          {
+            id: null,
+            type: null,
+            number: null,
+            ext: null,
+          },
+        ],
+        email: null,
+      },
+    ]
```
So when the author is archived, all the fields are set to `null`? 🤔 Is that just a Drupal thing, or..? How do you know?

brianseek replied:
I am not sure where this is happening to tell you the truth. I don't see where it is happening in next-build at a glance but I also can't imagine Drupal doing it like this. I'll dig around.

cvalarida reviewed (approved):
Looks great! Had a few questions, but :shipit:

laflannery commented:
@brianseek I wasn't clear on the ACs for the `<h2>` tags for Media and Download - but turns out I changed my mind anyway. I spoke to Michelle just now and can we have the `<h2>` but sized down to match an H3

`<h2 class="vads-u-font-size--h3">`

brianseek commented:
> @brianseek I wasn't clear on the ACs for the `<h2>` tags for Media and Download - but turns out I changed my mind anyway. I spoke to Michelle just now and can we have the `<h2>` but sized down to match an H3
> 
> `<h2 class="vads-u-font-size--h3">`

Pushed that update and updated screenshots.

laflannery reviewed (approved):
Thank you! Approved
